### DISCUSSION TITLE
Inconsistency on the logic to calculate monthly income

### DISCUSSION THREAD

**Discussion by igorfp25** [franciscobmacedo/remotefreelancept#62] [Q&A]
In the current code, the logic for the calculation of monthly income varies depending on the selected frequency, which can lead to inconsistent and confusing results:

1. When the frequency "Year" is selected, the monthly income is calculated by dividing the annual income by the number of months (using nrMonthsDisplay).
2. However, when "Day" is selected, monthly income is calculated as the daily rate multiplied by 22 days in a month (adjusted by a weight factor not relevant to this discussion).

This discrepancy creates an inconsistency. For example, with the "Day" frequency, a month is effectively treated as having 22 days, which sums up to 264 days over 12 months. And at the same time a year is considered to have 248 days.
This leads to a situation where multiplying the calculated monthly income by 12 does not match the calculated annual income, when the frequency "Day" is selected.

To keep the logic consistent, it would be preferable to use a unified approach across all frequencies. For example, if we adopt rule 1 for both cases, we would calculate annual income as daily rate * number of workdays in a year, and then determine monthly income by dividing annual income by the number of months.

This approach would eliminate the confusion and maintain consistent results, as the monthly income would either always represent the annual income divided by the number of months or always represent 22 days of work, but not both.

**Comment by franciscobmacedo** [comment]
I agree, I didn't even realize that they sum up to different numbers. 

Thanks for pointing it out! 

So, some scenarios to see if I understand it right:

**user input: 36000 €/year**
- year income would be 3600
- month income would be 3000 (36000/12)
- day income would be 136.36 (36000/12/22)

**user input: 3000 €/month**
- year income would be 3600 (3000*12)
- month income would be 3000 
- day income would be 136.36 (3000/22)

**user input: 136.36 €/day**
- year income would be 3600 (136.36*22*12)
- month income would be 3000 (136.36*22) 
- day income would be 136.36 


Would this solve the problem?

**Reply by bolt12** [reply]
I think the yearly salary is slightly different than the daily option. It all boils down to the contract nature. If you are paid a fixed monthly income it doesn't matter how many business days there are in a year, because at the end of the month you'll earn the same. However if your contract is hourly then each month has different business days and ends up having a different yearly salary each year

**Reply by franciscobmacedo** [reply]
Good point. Maybe it makes sense to add 2 input fields: one for how many working days per month you're working (on average) and one for how many months in a year. The defaults would be 22 and 12, respectively. 

A user that sets a daily income of 136€ can then say on average they work 22 days per month but only 11 months per year.

I think it's still relevant for a user that inputs a yearly salary to see how much it corresponds in daily or monthly income

**Reply by igorfp25** [reply]
> 
> **user input: 36000 €/year**
> 
> * year income would be 3600
> * month income would be 3000 (36000/12)
> * day income would be 136.36 (36000/12/22)
> 
> **user input: 3000 €/month**
> 
> * year income would be 3600 (3000*12)
> * month income would be 3000
> * day income would be 136.36 (3000/22)
> 
> **user input: 136.36 €/day**
> 
> * year income would be 3600 (136.36_22_12)
> * month income would be 3000 (136.36*22)
> * day income would be 136.36
> 
> Would this solve the problem?

This would solve the problem. 
However, I would personally prefer to have it solved in a different way, using business days in a year as a reference, rather than business days in a month. As follows: 

**user input: X €/year**
 
* year income would be X
 * month income would be X / number of months
* day income would be X / number of working days in a year
 
 **user input: Y €/month**
 
 * year income would be Y * number of months
 * month income would be Y
 * day income would be year income / number of working days in a year

 **user input: Z €/day**

* year income would be Z * number of working days in a year
* month income would be year income / number of months
* day income would be Z

This way we have the number of days in a year as reference, which let us have a better estimation considering all the holidays.
This would also facilitate the implementation of the new feature to choose the number of days off/vacation in a year (or maybe the number of working days in a year).


> It all boils down to the contract nature. If you are paid a fixed monthly income it doesn't matter how many business days there are in a year, because at the end of the month you'll earn the same.

As for this point, I understand that, in this case, the user would choose the frequency as "month". And the calculation would still make sense, in my point of view.
